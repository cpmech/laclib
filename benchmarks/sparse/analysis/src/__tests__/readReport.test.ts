import { readReport } from '../readReport';

describe('readReport', () => {
  it('reads bfwb62', () => {
    const rpt = readReport('mumps_bfwb62_metis_open_seq');
    expect(rpt).toMatchSnapshot();
  });

  it('fails due to wrong matrixName', () => {
    expect(() => {
      readReport('mumps_nada_metis_mpi1');
    }).toThrowError(
      "ENOENT: no such file or directory, open '../results/open/mumps_nada_metis_mpi1.json'",
    );
  });
});
