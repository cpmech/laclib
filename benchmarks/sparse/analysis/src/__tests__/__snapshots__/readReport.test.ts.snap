// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readReport reads bfwb62 1`] = `
Object {
  "MatrixName": "bfwb62",
  "MpiSize": 0,
  "NumberOfCols": 62,
  "NumberOfNonZeros": 202,
  "NumberOfRows": 62,
  "OmpNumThreads": 0,
  "Ordering": "metis",
  "Platform": "laclib",
  "SolverKind": "mumps",
  "Stats": Object {
    "ElapsedTimeNanoseconds": 2552,
    "ElapsedTimeString": "2.552µs",
    "NormInfA": 0.0001,
    "NormInfAx": 1,
    "NormInfDiff": 6.66133814775094e-16,
    "RelativeError": 6.66067208054288e-16,
  },
  "StepAnalyze": Object {
    "ElapsedTimeNanoseconds": 446633,
    "ElapsedTimeString": "446.633µs",
    "MemoryBytes": 505769984,
    "MemoryMiB": "482",
  },
  "StepFactorize": Object {
    "ElapsedTimeNanoseconds": 274147,
    "ElapsedTimeString": "274.147µs",
    "MemoryBytes": 15538995200,
    "MemoryMiB": "14,819",
  },
  "StepReadMatrix": Object {
    "ElapsedTimeNanoseconds": 100473,
    "ElapsedTimeString": "100.473µs",
    "MemoryBytes": 505503744,
    "MemoryMiB": "482",
  },
  "StepSolve": Object {
    "ElapsedTimeNanoseconds": 300100,
    "ElapsedTimeString": "300.1µs",
    "MemoryBytes": 15606104064,
    "MemoryMiB": "14,883",
  },
  "Symmetric": true,
  "TimeSolverNanoseconds": 891818,
  "TimeSolverString": "891.818µs",
}
`;
